### PR TITLE
Story 3

### PR DESCRIPTION
**Pull request by mattdyer1984**

### CODE REVIEW

File: src/Models/PlayerCardModel.php [hunk 1]
```diff
@@ -101,7 +107,18 @@ public function addNewCard(PlayerCard $newPlayer):bool {
...
             ':Attack' => $newPlayer->Attack,
-            ':Defence'=> $newPlayer->Defence
+            ':Defence' => $newPlayer->Defence
         ]);
     }
+
+    public function changeActivationStatus(int $id, int $Deleted)
+    {
+        $query = $this->db->prepare("UPDATE `PremierLeagueCards`
+        SET `Deleted` = $Deleted
```
Comment (ashleycoles) [inline src/Models/PlayerCardModel.php]: You should use parameter binding here to be safe, the data doesn't currently come from a user, but for the sake of a minute or two of extra work it's better to be cautious

File: src/Models/PlayerCardModel.php [hunk 2]
```diff
@@ -101,7 +107,18 @@ public function addNewCard(PlayerCard $newPlayer):bool {
...
+        WHERE `id` = $id
```
Comment (ashleycoles) [inline src/Models/PlayerCardModel.php]: Same with ID, use param binding to be safe

File: src/DeleteCard.php [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+            <form class='DeleteCardForm' action='RemoveCardPage.php' method='POST'>
+            <input type='hidden' name='card_id' value='{$card->id}'>
+            <input type='submit' name='delete' value='Delete' onclick='return confirmDelete()'>
+            </form>
+            ";
+        }
+    }
+
+    return $output;
+}
```
Comment (ashleycoles) [inline src/DeleteCard.php]: You should move this into its own separate file as well as adding a very exciting unit test for it

File: src/DeleteCard.php [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+function DeleteCard($id)
```
Comment (ashleycoles) [inline src/DeleteCard.php]: type hint

File: src/DeleteCard.php [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+{
+    $db = new PDO('mysql:host=db; dbname=Collection', 'root', 'password');
+    $db->setAttribute(PDO::ATTR_DEFAULT_FETCH_MODE, PDO::FETCH_ASSOC);
```
Comment (ashleycoles) [inline src/DeleteCard.php]: Now you're repeating this PDO connection code a number of times, you should make a function (in its own file) that returns $db

File: src/DeleteCard.php [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+require_once 'src/Models/PlayerCardModel.php';
+?>
+<script>
+    function confirmDelete() {
+        return confirm("Are you sure you want to delete this card?");
+    }
+</script>
+<?php
+
+function DisplayDataforCardDelete(array $allCards)
```
Comment (ashleycoles) [inline src/DeleteCard.php]: Needs a return type

File: src/DeleteCard.php [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+            <li>{$card->Club}</li>
+            <li>{$card->PositionName}</li>
+            <form class='DeleteCardForm' action='RemoveCardPage.php' method='POST'>
+            <input type='hidden' name='card_id' value='{$card->id}'>
+            <input type='submit' name='delete' value='Delete' onclick='return confirmDelete()'>
+            </form>
+            ";
+        }
+    }
+
```
Comment (ashleycoles) [inline src/DeleteCard.php]: There is a good chance you might not have any deleted cards, so worth checking for $allCards being empty and displaying message

File: src/displayCollectionDataList.php [hunk 8]
```diff
@@ -5,6 +5,7 @@ function displayCollectionDataList(array $allCards)
     $output = '';
 
     foreach ($allCards as $card) {
+      if ($card->Deleted == 0) {
```
Comment (ashleycoles) [inline src/displayCollectionDataList.php:8]: Instead of doing this, you should filter out deleted cards with your SQL query

**Review by ashleycoles — CHANGES_REQUESTED** [review]

File: src/DeleteCard.php (hunk 3 above)
Comment (mattdyer1984) [reply src/DeleteCard.php]: I knew you would say that!

**Review by ashleycoles — APPROVED** [review]

**Comment by mattdyer1984** [comment]
approved

**Pull request merged**